src/main/java/edu/gwu/csci6212/MinimumStepsChessKnight.java

```diff
@@ -0,0 +1,113 @@
+package edu.gwu.csci6212;
+
+
+import java.util.LinkedList;
+
+public class MinimumStepsChessKnight {
+
+    private int [][] board;
+    private int[] start;
+    private int[] goal;
+    private LinkedList<int[]> queue;
+    private boolean gameOver;
+    private int[][] knightMovements;
+    private int size;
+    private int totalSteps;
+
+    public MinimumStepsChessKnight(){
+        this.start = new int[3];
+        this.goal = new int[3];
+        this.queue = new LinkedList<>();
+        this.gameOver = false;
+
+        this.knightMovements = new int [][] {
+                //x y step
+                {1, 2, 1},
+                {1, -2, 1},
+                {2, 1, 1},
+                {2, -1, 1},
+                {-2, 1, 1},
+                {-2, -1, 1},
+                {-1, 2, 1},
+                {-1, -2, 1}
+        };
+
+    }
+
+    public void setUpBoard(int size, int[] start, int[] goal) throws Exception{
+        this.size = size;
+        board = new int[size][size];
+        this.start[0] = start[0];
+        this.start[1] = start[1];
+        this.start[2] = 0;
+        this.goal = goal;
+        if (outOfBoard()) throw new Exception("Position out of board");
+        board [start[0]][start[1]] = 1;
+        if (!isKnightInGoal()){
```

Comment:
What happens if the Knight is in the goal. I think we'd want to return 0 since the Knight had to make 0 moves to get to the destination, right?